class/practice_2.2/main.cpp

```diff
@@ -1,97 +1,119 @@
 #include <iostream>
 #include <fstream>
 #include <sstream>
+#include <array>
 #include <vector>
 #include <string>
+#include <optional>
 
 using namespace std;
 
+constexpr int WIDTH = 5;
+constexpr int HEIGHT = 5;
+
 struct Pixel {
     int r, g, b;
 
     bool operator==(const Pixel& other) const {
         return r == other.r && g == other.g && b == other.b;
     }
+
+    bool IsValid() const {
+        return (r >= 0 && r <= 255) && (g >= 0 && g <= 255) && (b >= 0 && b <= 255);
+    }
 };
 
-const int WIDTH = 5;
-const int HEIGHT = 5;
+using ImageRow = std::array<Pixel, WIDTH>;
 
-bool readImage(const string& filename, vector<vector<Pixel>>& image, const string& outputFilename) {
-    ifstream inFile(filename);
-    ofstream outFile(outputFilename);
-    if (!inFile) {
-        cerr << "Unable to open input file." << endl;
-        return false;
-    }
+class Image {
+public:
+    Image() : image{} {} 
 
-    string line;
-    int row = 0;
-    while (getline(inFile, line)) {
-        stringstream ss(line);
-        vector<Pixel> pixelRow;
-        for (int col = 0; col < WIDTH; ++col) {
-            Pixel pixel;
-            char comma;
-            ss >> pixel.r >> comma >> pixel.g >> comma >> pixel.b;
-            if (comma != ',' || pixel.r < 0 || pixel.r > 255 || pixel.g < 0 || pixel.g > 255 || pixel.b < 0 || pixel.b > 255) {
-                cerr << "Invalid input format at row " << row + 1 << ", column " << col + 1 << "." << endl;
+    bool readImage(const string& filename, const string& outputFilename) {
```

Comment:
we can think about better name for example readFromFile - it may be more convenient as we already inside the image class